In [55]:
import math

## Ejercicio 16: Trabajo en equipo.

En un grupo de 25 alumnos hay 13 chicas y 12 chicos. Se desean formar equipos de trabajo para realizar una investigación:

a) ¿Cuántos equipos de cinco personas pueden formarse?

Tenemos un total de 25 alumnos y si queremos realizar equipos de cinco personas cada uno usariamos la formula de las combinaciones, la cual es: 
$$
\begin{align}
    \binom{n}{k} = \frac{n!}{k!(n-k)!}
\end{align}
$$

Donde n = 25 y k = 5

$$
\begin{align}
    \binom{25}{5} &= \frac{25!}{5!(25-5)!}  &= 53130 \\ 
\end{align}
$$

In [56]:

equiposA = math.comb(25,5)

print(f'La cantidad de equipos que se pueden formar con 5 alumnos son: {equipos}')

La cantidad de equipos que se pueden formar con 5 alumnos son: 17160


b) ¿Cuántos equipos pueden formarse si debe haber una chica?

Primero debemos saber que si solo hay una chica en el equipo significa que los otros 4 integrantes seran chicos, sabiendo eso usamos la formula de las combinaciones para saber cuantas posibilidades hay de obtener a una chica en el equipo, y aparte saber cuantas hay de obtener a 4 chicos, para despues multiplicarlas entre si:
$$
\begin{align}
\binom{13}{1} = \frac{13!}{1!(13-1)!} = 13
\end{align}
$$
Son 13 posibilidades de que haya 1 chica en el equipo
$$
\begin{align}
\binom{12}{4} = \frac{12!}{4!(12-4)!} = 495
\end{align}
$$
Son 495 posibilidades de que haya 4 chicos en el equipo

$$
\begin{align}
    13 * 495 = 6435
\end{align}
$$

Ahora multiplicamos los resultados entre si para saber la cantidad de equipos posibles (los cuales son 6435)



In [57]:
equiposB = math.comb(13,1) * math.comb(12,4)
print(f'La cantidad de equipos que se pueden formar con 1 chica y 4 chicos son: {equipos}')

La cantidad de equipos que se pueden formar con 1 chica y 4 chicos son: 17160


c) ¿Cuántos equipos pueden formarse si debe haber dos chicas?

Siguiendo el mismo procedimiento del ejercicio anterior

Primero debemos saber que si solo hay dos chicas en el equipo significa que los otros 3 integrantes seran chicos, sabiendo eso usamos la formula de las combinaciones para saber cuantas posibilidades hay de obtener a dos chicas en el equipo, y aparte saber cuantas hay de obtener a 3 chicos, para despues multiplicarlas entre si:
$$
\begin{align}
\binom{13}{2} = \frac{13!}{2!(13-2)!} = 78
\end{align}
$$
Son 78 posibilidades de que haya 2 chicas en el equipo
$$
\begin{align}
\binom{12}{3} = \frac{12!}{3!(12-3)!} = 220
\end{align}
$$
Son 220 posibilidades de que haya 3 chicos en el equipo

$$
\begin{align}
    78 * 220 = 17160
\end{align}
$$
Ahora multiplicamos los resultados entre si para saber la cantidad de equipos posibles (los cuales son 17160)

In [58]:
equiposC = math.comb(13,2) * math.comb(12,3)
print(f'La cantidad de equipos que se pueden formar con 2 chicas y 3 chicos son: {equipos}')

La cantidad de equipos que se pueden formar con 2 chicas y 3 chicos son: 17160


d) Determina las probabilidades correspondientes a los incisos anteriores.

Para obtener las probabilidades correspondientes debemos saber que la probabilidad de obtiene dividiendo la cantidad de casos favorables entre la cantidad de casos totales:

$$
\begin{align}
    Probabilidad = \frac{Casos Favorables}{Casos Totales}
\end{align}
$$

Ya con eso podemos obtener las probabilidades correspondientes

Para el inciso b:

$$
\begin{align}
    Probabilidad = \frac{6435}{53130} =
\end{align}
$$

Y para el inciso c:

$$
\begin{align}
    Probabilidad = \frac{17160}{53130} =
\end{align}
$$

In [59]:
ProbabilidadB = equiposB/equiposA
print(f'La probabilidad de que salga un equipo con una chica y cuatro chicos es de: {round(ProbabilidadB*100,2)} %')
ProbabilidadC = equiposC/equiposA
print(f'La probabilidad de que salga un equipo con dos chicas y tres chicos es de: {round(ProbabilidadC*100,2)} %')


La probabilidad de que salga un equipo con una chica y cuatro chicos es de: 12.11 %
La probabilidad de que salga un equipo con dos chicas y tres chicos es de: 32.3 %


## Ejercicio 17: Control de calidad.

Se tiene una caja con 24 latas y se sabe que dos están contaminadas. Se van a seleccionar tres latas al azar para someterlas a una prueba de control de calidad, es decir, para medir los estándares de calidad de la empresa.

a) ¿Cuántas combinaciones de tres latas pueden hacerse?

Aquí nos pide obtener la cantidad de combinaciones que se pueden obtener con tres latas, entonces usamos la formúla de las combinaciones: 

$$
\begin{align}
    \binom{n}{k} = \frac{n!}{k!(n-k)!}
\end{align}
$$

Donde n = 24 y k = 3

Donde n = 25 y k = 5

$$
\begin{align}
    \binom{24}{3} &= \frac{24!}{3!(24-3)!}  &= 2024 \\ 
\end{align}
$$

In [60]:
latasA = math.comb(24,3)
print(f'La cantidad de combinaciones que se pueden hacer de tres latas: {tresLatas}')

La cantidad de combinaciones que se pueden hacer de tres latas: 2024


b)¿Cuál es la probabilidad de que se seleccione una lata contaminada para la prueba?

En este ejercicio podemos deducir que de las tres latas a las que se les realizan la prueba una sería contaminada y las otras dos serian normales, entonces debemos obtener la cantidad de combinaciones donde esos casos sean favorables.

Obtenemos la cantidad de combinaciones que se puedan realizar donde se obtenga una lata de las dos contaminadas

$$
\begin{align}
\binom{2}{1} = \frac{2!}{1!(2-1)!} = 2
\end{align}
$$
Son 2 posibilidades de que haya una lata contaminada en la prueba

Ahora hacemos lo mismo para obtener la cantidad de combinaciones para las latas no contaminadas que serian 22 para obtener solo 2

$$
\begin{align}
\binom{22}{2} = \frac{22!}{2!(22-2)!} = 231
\end{align}
$$
Son 231 posibilidades de que haya dos latas no contaminadas

$$
\begin{align}
    2 * 231 = 462
\end{align}
$$

Ahora multiplicamos los resultados entre si para saber la cantidad de equipos posibles (los cuales son 462)

Y para obtener la probabilidad aplicamos la siguiente formula: 


$$
\begin{align}
    Probabilidad = \frac{Casos Favorables}{Casos Totales}
\end{align}
$$

Donde los casos favorables son 462 y los casos totales son 2024


$$
\begin{align}
    Probabilidad = \frac{462}{2024} = 0.2283
\end{align}
$$

La probabilidad es de 22.83%



In [61]:
latasB = math.comb(2,1) * math.comb(22,2)
ProbabilidadB = latasB / latasA
print(f'La probabilidad de obtener una lata contaminada en la prueba es de: {round(ProbabilidadB*100,2)} %')


La probabilidad de obtener una lata contaminada en la prueba es de: 22.83 %


c) ¿Cuál es la probabilidad de que se seleccione al menos una lata contaminada para la prueba?

Para este ejercicio debemos hacer enfasis en el punto donde dice al menos una lata pues esto nos indica que existe la posibilidad de que haya dos latas contaminadas, como ya obtuvimos anteriormente la probabilidad para una lata ahora solo obtenemos el de dos latas y lo sumamos:

btenemos la cantidad de combinaciones que se puedan realizar donde se obtenga dos latas de las dos contaminadas

$$
\begin{align}
\binom{2}{2} = \frac{2!}{2!(2-2)!} = 1
\end{align}
$$
Hay 1 posibilidad de que haya dos latas contaminadas en la prueba

Ahora hacemos lo mismo para obtener la cantidad de combinaciones para las latas no contaminadas que serian 22 para obtener solo 1

$$
\begin{align}
\binom{22}{1} = \frac{22!}{1!(22-1)!} = 22
\end{align}
$$
Son 22 posibilidades de que haya una lata no contaminadas

$$
\begin{align}
    1 * 22 = 22
\end{align}
$$

Aplicamos la formula de la probabilidad

$$
\begin{align}
    Probabilidad = \frac{22}{2024} = 0.01
\end{align}
$$

La probabilidad es de 1.08%

Y sumandola a la probabilidad anterior de 22.83% nos da igual a 23.91% en total


In [65]:
latasC = math.comb(2,2) * math.comb(22,1)
ProbabilidadC1 = latasC / latasA
ProbabilidadC2 = ProbabilidadC1+ProbabilidadB
print(f'La probabilidad de obtener al menos 1 lata contaminada en la prueba es de: {round(ProbabilidadC2*100,2)} %')


0.010869565217391304
La probabilidad de obtener al menos 1 lata contaminada en la prueba es de: 23.91 %


d) ¿Y la probabilidad de que no se elijan latas contaminadas para la prueba?

Ahora como para este punto tenemos los casos contrarios a este inciso que son cualquier caso donde exista una lata contaminada, lo unico que tendriamos que hacer es restarle la probabilidad anterior que es 23.91 a 100 y tendriamos nuestro resultado: 

$$ 100 - 23.91 =  76.087 $$

In [70]:

ProbabilidadD = 1-ProbabilidadC2

print(f'La probabilidad de no obtener ninguna lata contaminada en la prueba es de: {round(ProbabilidadD*100,2)} %')

La probabilidad de no obtener ninguna lata contaminada en la prueba es de: 76.09 %
